lib/posix/pthread_mutex.c

```diff
@@ -22,13 +24,14 @@ static const pthread_mutexattr_t def_attr = {
 
 static int acquire_mutex(pthread_mutex_t *m, k_timeout_t timeout)
 {
-	int rc = 0, key = irq_lock();
+	int rc = 0;
+	k_spinlock_key_t key = k_spin_lock(&_z_pthread_spinlock);
```

Comment:
I believe this will break the `PTHREAD_MUTEX_RECURSIVE` use case and result in deadlock - at least that's what [the docs](https://docs.zephyrproject.org/latest/reference/kernel/smp/smp.html?#spinlocks) lead me to believe.

@evgeniy-paltsev - Can you please supply a test case that verifies expected behaviour?